xarray/conventions.py

```diff
@@ -577,23 +604,28 @@ def stackable(dim):
                         )
                     del var_attrs[attr_name]
 
-    if decode_coords and "coordinates" in attributes:
+    if (
+        decode_coords
+        and "coordinates" in attributes
+        and isinstance(attributes["coordinates"], str)
+    ):
         attributes = dict(attributes)
-        coord_names.update(attributes.pop("coordinates").split())
+        crds = attributes.pop("coordinates")
+        coord_names.update(crds.split())
```

Comment:
Might be a functionality change? Should this raise an error instead if it's not a string?

Reply:
Yes it would be good to raise a nice error. `"coordinates"` is expected to be a string with variable names separated by spaces: http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#attribute-appendix